### DISCUSSION TITLE
What does "Extended text selection keys" in conhost settings do?

### DISCUSSION THREAD

**Discussion by bzzrak** [microsoft/terminal#14973] [Q&A] [answered]
I'm talking about this: (the second-to-last checkbox)
https://miro.medium.com/v2/1*kcH73rMwnC88hm2-Rz03qQ.png

[The documentation](https://github.com/microsoft/terminal/blob/main/doc/ConsoleHostSettings.md) says "Whether to allow the use of extended edit keys or not", which doesn't explain what those extended keys are. Some websites say that it allows shift+arrow keys and Ctrl+A to select, but that works fine with that option off as well.
I tried looking at the code, but I don't know C++ so I really can't make any sense of it.

So does it even do anything?

**Comment by j4james** [comment, marked as answer]
I'm fairly certain that option does nothing. It's loaded into a global variable named `g_fEditKeys`, but if you search the source, you'll see that variable is not really used anywhere. For more information on the origin of the option, see the comment here: https://github.com/microsoft/terminal/issues/809#issuecomment-492855406.

**Reply by zadjii-msft** [reply]
Yea, that's about right. I believe extended edit keys was a prototypical form of being able to rebind what keys do which "actions" in conhost, long before the terminal existed... but it was never documented, we never understood it, and it didn't seem particularly well understood. I remember us removing that codepath probably 5 years ago now because it was _literally never used_. And we obviously went in an entirely different direction with keybindings in the Terminal.
